util/reggen/reg_block.py

```diff
@@ -507,3 +569,86 @@ def has_internal_shadowed_reg(self) -> bool:
                 return True
 
         return False
+
+    def apply_alias(self, alias_block: object, where: str) -> None:
+        """
+        Validates alias description and overrides values in this block.
+
+        This updates the overridable register and field attributes with the
+        alias values and ensures that all non-overridable attributes have
+        identical values.
+        """
+
+        if not isinstance(alias_block, RegBlock):
+            raise ValueError('alias_block is not of type RegBlock')
+
+        # Before doing anything, check that the new alias names do not exist.
+        intersection = (alias_block.name_to_flat_reg.keys() &
+                        self.name_to_flat_reg.keys())
+        if intersection:
+            raise ValueError('Alias register names {} are not unique in alias '
+                             ' {}'.format(list(intersection), where))
+
+        # Loop over registers, validate the structure and update the generic
+        # register data structure. Since the internal register
+        # lists "registers", "flat_regs", "all_regs", "type_regs"
+        # and "entries" use references to the reg objects, everything stays in
+        # sync - except when multiregs are involved. To that end, another
+        # update loop over multiregs is required further below.
+        for alias_reg in alias_block.registers:
+            # First, check existence of the register to be aliased
+            if alias_reg.alias_target is None:
+                raise ValueError('No alias target register defined for '
+                                 'alias name {} in {}'
+                                 .format(alias_reg.name, where))
+
+            target = alias_reg.alias_target.lower()
+            if target not in self.name_to_flat_reg:
+                raise ValueError('Aliased target register {} with alias '
+                                 'name {} does not exist in reg '
+                                 'block {} ({}).'
+                                 .format(target,
+                                         alias_reg.name,
+                                         self.name,
+                                         where))
+
+            # This is the register we want to alias over. Check that the
+            # non-overridable attributes match, and override the attributes.
+            reg = self.name_to_flat_reg[target]
+            reg.apply_alias(alias_reg, where)
+
+        # Build a local index of all multiregs. We don't store this in the
+        # class since it is only used once here.
+        name_to_multiregs = {}
+        for mr in self.multiregs:
+            name_to_multiregs[mr.name.lower()] = mr
```

Comment:
You can do this with a single comprehension, I think: `name_to_multiregs = {mr.name.lower(): mr for mr in self.multiregs}`.